### PR TITLE
[AGENDA-147] Improve occupations validation

### PR DESCRIPTION
**Pull request by uasouz**
## What does this PR do?

Fixes # (issue)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How should this be tested?

- [ ] Vacancy concurrency need to be validate
- [ ] No more than 3 persons can be able to schedule in a time range

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] My changes generate no new warnings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pages/api/book/pac-event.ts [hunk 1]
```diff
@@ -398,17 +398,23 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-      horarioInicioAtendimento: dayjs(reqBody.start).toDate(),
-      //horarioFinalAtendimento: dayjs(reqBody.start).toDate(),
+      horarioInicioAtendimento: {
+        gte: dayjs(reqBody.start).utc().toDate(),
+        lt: dayjs(reqBody.end).utc().toDate(),
+      },
+      horarioFinalAtendimento: {
+        gt: dayjs(reqBody.start).utc().toDate(),
+        lte: dayjs(reqBody.end).utc().toDate(),
+      },
```
Comment (armr-dev) [inline pages/api/book/pac-event.ts:410]: This validation might break something with the workaround that I did with the timezone. It shouldn't, but we should test it just to guarantee. 

@LuisHMA please, make sure this is tested properly.

File: pages/api/book/pac-event.ts [hunk 2]
```diff
@@ -421,15 +427,15 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
-    gmtTimezone: -4,
-    status: 3,
-  });
+  // log.debug(occupationsOnDate, {
+  //   horarioInicioAtendimento: dayjs(reqBody.start).toDate(),
+  //   horarioFinalAtendimento: dayjs(reqBody.end).toDate(),
+  //   pac: customInputValues.pac,
+  //   gmtTimezone: -4,
+  //   status: 3,
+  // });
```
Comment (armr-dev) [inline pages/api/book/pac-event.ts:436]: 🔥🔥🔥🔥🔥?

File: pages/api/book/pac-event.ts [hunk 3]
```diff
@@ -398,17 +398,23 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+        lte: dayjs(reqBody.end).utc().toDate(),
+      },
       pac: customInputValues.pac,
       gmtTimezone: -4,
       status: 3,
     },
   });
 
-  // const vacanciesCount = await prisma.$queryRaw(Prisma.sql`select (3 - count(*)) as vacancies
+  // const occupationsOnDate = await prisma.$queryRaw(Prisma.sql`select (3 - count(*)) as vacancies
```
Comment (armr-dev) [inline pages/api/book/pac-event.ts:417]: 🔥🔥🔥🔥?

File: pages/api/book/pac-event.ts [hunk 4]
```diff
@@ -421,15 +427,15 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+  //   horarioInicioAtendimento: dayjs(reqBody.start).toDate(),
+  //   horarioFinalAtendimento: dayjs(reqBody.end).toDate(),
+  //   pac: customInputValues.pac,
+  //   gmtTimezone: -4,
+  //   status: 3,
+  // });
 
-  if (vacanciesCount > 2) {
+  if (occupationsOnDate > 2) {
     console.log({ message: "booking.noAvailability" });
```
Comment (armr-dev) [inline pages/api/book/pac-event.ts:439]: 🔥 or change it to `log.debug()`

**Review by armr-dev — APPROVED** [review]

**Review by LuisHMA — APPROVED** [review]

**Review by EmanSilva — APPROVED** [review]

**Pull request merged**